src/esbmc/bmc.cpp

```diff
@@ -810,68 +839,56 @@ smt_convt::resultt bmct::multi_property_check(
-            options.get_bool_option("make-assert-false") ||
-            options.get_bool_option("add-false-assert"))
-          {
-            if(claim.claim_msg.find("Instrumentation") != std::string::npos)
-              tracked_instrument++;
-          }
+          //TODO: this can still be annoying when we unind for many times
+          // e.g. '--unwind 100' will show 1 counterexample and 99 'skip's
+          // Maybe we should use log_debug instead, both in slicer.run and multi_property_check
         }
-        // TODO: This is the place to store into a cache
       }
-      catch(...)
-      {
-        log_debug("multi-property", "Failing Fast");
-      }
-    };
+    }
+    catch(...)
+    {
+      log_debug("multi-property", "Failing Fast");
+    }
+  };
 
-  // PARALLEL
-  if(options.get_bool_option("parallel-solving"))
-  {
-    /* NOTE: I would love to use std::for_each here, but it is not giving
-       * the result I would expect. My guess is either compiler version
-       * or some magic flag that we are not using.
-       *
-       * Nevertheless, we can achieve the same results by just creating
-       * threads.
-       */
-
-    // TODO: Running everything in parallel might be a bad idea.
-    //       Should we also add a thread pool?
-    std::vector<std::thread> parallel_jobs;
-    for(const auto &i : jobs)
-      parallel_jobs.push_back(std::thread(job_function, i));
-
-    // Main driver
-    for(auto &t : parallel_jobs)
-    {
-      t.join();
-    }
-    // We could remove joined jobs from the parallel_jobs vector.
-    // However, its probably not worth for small vectors.
-  }
-  // SEQUENTIAL
-  else
-    std::for_each(std::begin(jobs), std::end(jobs), job_function);
+  std::for_each(std::begin(jobs), std::end(jobs), job_function);
 
-  if(
-    options.get_bool_option("make-assert-false") &&
-    !(options.get_bool_option("goto-coverage") ||
-      options.get_bool_option("add-false-assert")))
+  // For coverage
+  if(is_goto_cov)
   {
     int total = goto_coveraget().get_total_instrument();
+    int tracked_instance = reached_mul_claims.size();
+
     if(total)
     {
       log_success("\n[Coverage]\n");
-      log_result("  Total Asserts: {}", total);
-      log_result("  Reached Asserts: {}", tracked_instrument);
-      log_result("  Coverage: {}%", tracked_instrument * 100.0 / total);
+      log_result("Total Asserts: {}", total);
+      log_result("Total Assertion Instances: {}", total_instance);
+    }
+
+    // show claims
+    if(options.get_bool_option("goto-coverage-claims"))
+    {
+      // reached claims:
+      log_result("Reached Assertions Instances: ");
+      for(const auto &claim : reached_mul_claims)
+      {
+        log_status("  {}", claim);
+      }
     }
+
+    //TODO: show unreached claims
+
+    if(total_instance != 0)
+      log_result("Coverage: {}%", tracked_instance * 100.0 / total_instance);
+    else
+      log_result("Coverage: 0%");
```

Comment:
We may want to relabel this. "Coverage" usually has a quite specific meaning, which differs from what is being tested here by replacing asserts by assert(0) and testing each one separately. @mikhailramalho's comment https://github.com/esbmc/esbmc/pull/1337#issuecomment-1717900619 is very relevant to the original meaning of coverage:

>  We can easily check if an instruction is reachable by checking if its guard
is true.

Reply:
I agree. It is no longer the original meaning of coverage.

Maybe we should change it to something like `Assertion Instance Coverage`, which is reached_instance/total_instance. The total assertion instances include the assert inside the source file, the unwinding asserts, the claims inserted during the goto-check and so on.

Reply:
I agree, with this explanation also given in the help message for the `--goto-coverage` option that would make it more clear.